projects/ComposeConstraintLayout/app/src/main/java/com/example/constraintlayout/MotionCarouselDemos.kt

```diff
@@ -0,0 +1,465 @@
+                      height: 200,
+                      end: ['parent','end', 8],
+                      centerVertically: 'parent',
+                      custom: {
+                        mainColor: '#0000FF'
+                      }  
+                    } 
+        """}
+    val backwardCs = remember {"""
+                    label: {
+                      start: ['parent', 'start', 8],
+                      top: ['parent', 'top', 8]
+                    },
+                    card0: {
+                      width: 100,
+                      height: 200,
+                      start: ['parent','start', 8],
+                      centerVertically: 'parent',
+                      custom: {
+                        mainColor: '#0000FF'
+                      }  
+                    }, 
+                    card1: {
+                      width: 150,
+                      height: 250,
+                      centerHorizontally: 'parent',
+                      centerVertically: 'parent',
+                      rotationX: 5,
+                      //rotationY: 360,
+                      custom: {
+                          mainColor: '#FF0000',
+                          main: 1.0
+                      }
+                    },
+                    card2: {
+                      width: 100,
+                      height: 200,
+                      end: ['parent','end', 8],
+                      centerVertically: 'parent',
+                      custom: {
+                        mainColor: '#0000FF'
+                      }  
+                    },
+                    card3: {
+                      width: 100,
+                      height: 200,
+                      start: ['parent','end', 8],
+                      centerVertically: 'parent',
+                      custom: {
+                        mainColor: '#0000FF'
+                      }  
+                    },
+                    card4: {
+                      width: 100,
+                      height: 200,
+                      start: ['card3','end', 8],
+                      centerVertically: 'parent',
+                      custom: {
+                        mainColor: '#0000FF'
+                      }  
+                    },             
+        """}
+    val motionScene = MotionScene("""{
+                Header: {
+                  exportAs: 'carousel'
+                },
+                ConstraintSets: {
+                  start: { $startCs },
+                  forward: { $forwardCs },
+                  backward: { $backwardCs }
+                },
+                Transitions: {
+                  forward: {
+                    from: 'start',
+                    to: 'forward',
+                  },
+                  backward: {
+                    from: 'start',
+                    to: 'backward',
+                  }
+            }""")
+    MotionCarousel(motionScene, 2, 5, content = content, showSlots = false)
```

Comment:
These demos are probably good from a look point of view.
But they are difficult to follow hence bad from an educational point of view.

Can you make another demo that is as compact and self contained as possible?